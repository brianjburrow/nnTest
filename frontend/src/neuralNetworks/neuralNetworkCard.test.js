// tests based on Jobly App (Springboard)
import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import NeuralNetworkCard from "./neuralNetworkCard";
import { UserProvider } from "../testUtils";

let TEST_LAYER = [
    { type: "lstm", params: { units: 25, activation: 'relu', inputShape: `3` } },
    { type: "dense", params: { units: 1 } }
]

it("renders without crashing", function () {
    render(<NeuralNetworkCard layerList = {TEST_LAYER}/>);
});

it("matches snapshot when logged in", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <UserProvider>
            <NeuralNetworkCard layerList = {TEST_LAYER}/>
          </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <UserProvider currentUser={null}>
            <NeuralNetworkCard layerList = {TEST_LAYER}/>
          </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  