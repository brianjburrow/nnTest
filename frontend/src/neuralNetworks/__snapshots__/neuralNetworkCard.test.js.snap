// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div
    class="NeuralNetworkCard card col-6 "
  >
    <div
      class="card-body"
    >
      <div>
        <div>
          Layer #0
          <p>
            lstm
          </p>
          <ul>
            <li>
              units:25
            </li>
            <li>
              activation:relu
            </li>
            <li>
              inputShape:3
            </li>
          </ul>
          <button>
            Insert Layer
          </button>
          <button>
            Delete Layer
          </button>
        </div>
      </div>
      <div>
        <div>
          Layer #1
          <p>
            dense
          </p>
          <ul>
            <li>
              units:1
            </li>
          </ul>
          <button>
            Insert Layer
          </button>
          <button>
            Delete Layer
          </button>
        </div>
      </div>
    </div>
    <button
      class="btn btn-primary"
    >
      Train Neural Network
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div
    class="NeuralNetworkCard card col-6 "
  >
    <div
      class="card-body"
    >
      <div>
        <div>
          Layer #0
          <p>
            lstm
          </p>
          <ul>
            <li>
              units:25
            </li>
            <li>
              activation:relu
            </li>
            <li>
              inputShape:3
            </li>
          </ul>
          <button>
            Insert Layer
          </button>
          <button>
            Delete Layer
          </button>
        </div>
      </div>
      <div>
        <div>
          Layer #1
          <p>
            dense
          </p>
          <ul>
            <li>
              units:1
            </li>
          </ul>
          <button>
            Insert Layer
          </button>
          <button>
            Delete Layer
          </button>
        </div>
      </div>
    </div>
    <button
      class="btn btn-primary"
    >
      Train Neural Network
    </button>
  </div>
</DocumentFragment>
`;
